### ISSUE TITLE
improve location feature?

### ISSUE THREAD

**Issue by rickyrsyah** [pedroslopez/whatsapp-web.js#183]
right now, when type of message is location. will got number value's of longitude, number value's of latitude, preview map jpeg file in base64, and value of description if exist and undefined if live location.
is possible to get fresh latitude and longitude when share duration is not expired and state of live location still sharing?

**Comment by pedroslopez** [comment]
Yeah, live locations are not handled at the moment. It's definitely something I can look into supporting :)

**Comment by Aliyss** [comment]
Prerequisites: Create a live location from phone. Send it to a chat.

```js
let livelocationModel = mR.modules[36839].default.models[0]

let livelocationParticipant = mR.modules[36839].default.models[0].participants.models[0]
```

**Comment by HugoYG** [comment]
Do u have a complete example? I didn't get it.

> 
> 
> Prerequisites: Create a live location from phone. Send it to a chat.
> 
> ```js
> let livelocationModel = mR.modules[36839].default.models[0]
> 
> let livelocationParticipant = mR.modules[36839].default.models[0].participants.models[0]
> 
> // Register new LiveLocations
> mR.modules[36839].default.on('all', (x, y, z) => console.log(x, y, z))
> ```

**Comment by Aliyss** [comment]
@HugoYG 

It's sample code to use live locations. I commented it in case someone wanted to make a pull request themselves.

I haven't gotten around to integrating it within the code.
Let me see how I can actually add it within the package. Totally forgot about it.

I'll ping you once the pull request is done.

**Comment by josemontiel95** [comment]
@Aliyss 
Any luck adding it within the package?

**Comment by cezarlacatus** [comment]
Hello, is it possible to realtime read live location lat and long from a message you receive?

I want to make a PWA and get users location through whatsapp as PWA's are not allowed to get Location in Background, so the user can just share their live location through whatsapp at a service number.

**Comment by josemontiel95** [comment]
I'm still Interested on this feature. Anyone else?

**Comment by Aliyss** [comment]
Afaik the last time I tested I couldn't get WhatsApp Web to show the live location itself.

@josemontiel95 Do you see the live location in WhatsApp Web?

**Comment by tuyuribr** [comment]
stale

**Comment by syonfox** [comment]
<img width="1166" height="351" alt="Image" src="https://github.com/user-attachments/assets/33500162-2c5c-4a8f-acde-552237f0ba84" />

Yup in whatsapp web it just says unavailable.
